general/mm.cpp

```diff
@@ -171,6 +171,7 @@ void *mm::Erase(void *ptr)
 static inline bool MmDeviceIniFilter(void)
 {
    if (!config::UsingMM()) { return true; }
+   if (config::GetMode() == config::HostMode()) { return true; }
```

Comment:
I'm not sure about this.
This function filters out the `ptr` computation, depending on the initial config (`MFEM_USE_MM`) and the device status (has it been initialized yet).
If we want to get the right pointer before hitting a kernel, we need to move the data if necessary (after a 'mode' switch).
If we just `return` there, I don't think there is going to be some transfers back from the GPU.

Reply:
Oh I see. That is a good point. What about an option for disabling the device? For example, I only want to apply Okina to one portion of the code (say the middle). After the loops have been executed it would be nice if the code just behaved as though the GPU had not been enabled. This would enable developers to adopt Okina incrementally (maybe do a massive pull of data from device to host when disabling).

Reply:
Actually, I think what may be more semantically correct are the following options: 
`config::DisableTracking()`
`config::EnableTracking()` 

With the caveat that when we disable tracking, we assume that the data has been pulled to the host and we will only compute on the host henceforth. 
I'm trying it out in a code right now, I think it may make it easier for integration.